packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,324 @@
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
+
+```HTML
+<template slot="${x => x.columnInternals.uniqueId}">
+    <slot
+        ${slotted('mappings')}
+        name="mapping">
+    </slot>
+    <span class="header-content">
+        <slot></slot>
+    </span>
+</template>
+```
+
+Cell view:
+
+The cell view relies on the matched mapping to provide a template to render.
+
+```HTML
+${repeat(x => (x.column as TableColumnMapping).mappings,
+    html<Mapping, TableColumnMappingCellView>`
+        ${when((x, c) => x.typedKey === (c.parent as TableColumnMappingCellView).cellRecord.value,
+            html<Mapping>`${x => x.cellViewTemplate}`
+        )}
+    `
+)}
+```
+
+Group header view:
+
+Similarly, the group header view relies on the matched mapping to provide a template to render.
+
+Note the following requires that `TableColumnMappingGroupHeaderView` has a reference to the column with which it is associated. This is needed to enumerate the column's mapping elements.
+
+```HTML
+${repeat(x => (x.column as TableColumnMapping).mappings,
+    html<Mapping, TableColumnMappingHeaderView>`
+        ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+            html<Mapping>`${x => x.groupHeaderViewTemplate}`
+        )}
+    `
+)}
+```
+
+#### `nimble-mapping-*`
+
+```HTML
+<template slot="mapping"></template>
+```
+
+#### `nimble-mapping-icon`
+
+Cell view template:
+
+```HTML
+<${this.icon}
+    title="${x => x.label}"
+    aria-label="${x => x.label}"
+    severity="${x => x.severity}">
+</${this.icon}>
+```
+
+Group header view template:
+
+```HTML
+<${this.icon}
+    title="${x => x.label}"
+    aria-label="${x => x.label}"
+    severity="${x => x.severity}">
+</${this.icon}>
+<span
+    ${ref('span')}
+    @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.label && x.span!.offsetWidth < x.span!.scrollWidth;
+    }}"
+    @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+    title=${x => (x.isValidContentAndHasOverflow ? x.label : null)}>
+    ${x => x.label}
+</span>`;
+```
+
+### Grouping
+
+Grouping will be based on the record value. The grouping header will display the rendered icon/spinner/text. In the case of an icon/spinner, it will also be followed by the `label` text. This will disambiguate cases where multiple record values map to the same icon (assuming the labels are different). Text in a grouping header will be ellipsized and gain a tooltip if there is not enough room to display it all.
+
+### Sorting
+
+Sorting will be based on the record value. For icons, if multiple values map the to same icon, it is possible that sorting will result in the instances of a certain icon not being all together in one span of rows. The user may think they should be all together, but this is a corner case that we can't/shouldn't do anything about.
```

Comment:
Could you think through which sorting mode (e.g. locale-aware or basic) we'd use for each data type? 

I wonder if users would find it confusing that we sort by record value if the mapped string values would produce a different sort order. Is there any way we could provide a custom sorting mode for mapped string values?

Reply:
Please read through the update I've made to that section of the spec. TLDR: I think we should just use a basic sort regardless of the data type.

> Is there any way we could provide a custom sorting mode for mapped string values?

No, TanStack only supports sorting by a value in the record, which the mapped value is not. Also, if we want to support server-side sort, that also conflicts with sorting by a mapped value.